be/src/io/cache/block_file_cache.cpp

```diff
@@ -1648,11 +1666,69 @@ void BlockFileCache::check_disk_resource_limit() {
     }
 }
 
+void BlockFileCache::check_need_evict_cache_in_advance() {
+    if (_storage->get_type() != FileCacheStorageType::DISK) {
+        return;
+    }
+
+    std::pair<int, int> percent;
+    int ret = disk_used_percentage(_cache_base_path, &percent);
```

Comment:
it may not work if the disk is not fully used by doris, e.g. we set file_cache capacity to 100GB how ever the disk capacity is 1000GB.